lib/template-mixins.js

```diff
@@ -384,18 +399,19 @@ module.exports = function (fields, options) {
                                 month: autocomplete + '-month',
                                 year: autocomplete + '-year'
                             };
+                            delete field.autocomplete;
                         }
 
                         var parts = [],
                             dayPart, monthPart, yearPart;
 
                         if (isExact) {
-                            dayPart = compiled['partials/forms/input-text-group'].render(inputText.call(this, key + '-day', { pattern: '[0-9]*', min: 1, max: 31, maxlength: 2, hintId: key + '-hint', date: true, autocomplete: autocomplete.day }));
+                            dayPart = compiled['partials/forms/input-text-group'].render(inputText.call(this, field, key + '-day', { pattern: '[0-9]*', min: 1, max: 31, maxlength: 2, hintId: key + '-hint', date: true, autocomplete: autocomplete.day }));
```

Comment:
For settings like 'labelClassName' to be picked up the 'field' argument should be the day part of the date - this.options.fields[key + '-day']. Same for month and year below.